src/AtlasAdmin.ts

```diff
@@ -0,0 +1,172 @@
+// @ts-ignore
+import Logger from 'basic-logger';
+import { request, RequestOptions } from 'urllib';
+import { SearchIndex } from './data/atlas-search-index';
+import { SearchIndexResponse, IndexMappings } from './data/atlas-types';
+import { Taxonomy } from './SearchIndex';
+
+const DEFAULT_ATLAS_API_OPTIONS: RequestOptions = {
+  headers: {
+    'content-type': 'application/json',
+  },
+  dataType: 'json',
+};
+
+const log = new Logger({
+  showTimestamp: true,
+});
+
+const CLUSTER_NAME = process.env['CLUSTER_NAME'] || 'Search';
+const COLLECTION_NAME = process.env['COLLECTION_NAME'] || 'documents';
+const GROUP_ID = process.env['GROUP_ID'] || '';
+const DB = process.env['ATLAS_DATABASE'] || 'search';
+const SEARCH_INDEX = 'default';
+
+/**
+ * Manager is intended to keep Atlas Search Index in sync across environments
+ * Index should be edited in ../data directory and expect to be updated on deploy
+ */
+
+export class AtlasAdminManager {
+  private defaultHeaders: RequestOptions;
+  private baseUrl: string;
+
+  constructor(publicApiKey: string, privApiKey: string) {
+    // set base headers
+    this.defaultHeaders = DEFAULT_ATLAS_API_OPTIONS;
+    this.defaultHeaders['digestAuth'] = `${publicApiKey}:${privApiKey}`;
+    // set base url
+    this.baseUrl = `https://cloud.mongodb.com/api/atlas/v1.0/groups/${GROUP_ID}/clusters/${CLUSTER_NAME}/fts/indexes`;
+  }
+
+  async patchSearchIndex(taxonomy: Taxonomy) {
+    log.info('patchSearchIndex');
+    try {
+      const index = await this.findSearchIndex(DB, COLLECTION_NAME, SEARCH_INDEX);
+      if (!index) {
+        return this.createSearchIndex(SearchIndex, taxonomy);
+      }
+      return this.updateSearchindex(index['indexID']!, SearchIndex, taxonomy);
```

Comment:
Sorry to be a stickler about this, lol, but I think we should check if `indexID` is defined instead of using the non-null assertion operator.

If you do something like

```ts
.
.
.
// index.indexID is of type string | undefined before this point

if (!index.indexID) {
 
 throw new Error('IndexID is not defined');
}

// index.indexID is of type string past this point
.
.
.
```

Reply:
im gonna update the response to always have this instead of making a conditional

https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Atlas-Search/operation/listAtlasSearchIndexes

not sure why the indexId would not be `required` under the response, but the docs show the response will return indexid: https://www.mongodb.com/docs/atlas/atlas-search/tutorial/create-index/